app/models/value_sorter.rb

```diff
@@ -3,8 +3,12 @@
 module ValueSorter
   class << self
     def sort(values, locale: "en")
-      if values.first.position.present?
-        values.sort_by(&:position)
+      return values if values.length <= 1
+
+      if values.any? { _1.position.present? }
+        values.sort_by do |v|
+          v.position || values.map(&:position).compact.max + 1
+        end
```

Comment:
Should this be `values.all?`? That would let us keep the sort simple as well.

When do we have a case of something being manually sorted but missing a position in practice?